### ISSUE TITLE
Unable to log anything in UWP C# project 

### ISSUE THREAD

**Issue by karolys** [NLog/NLog#1264]
Attempting to integrate NLog into my UWP project to run on a Windows 10 system and I'm unable to get any sort of output from NLog. Neither my custom file target nor the internal logging seem to be writing out anything. 

[nlogTestApp_UWP.zip](https://github.com/NLog/NLog/files/145620/nlogTestApp_UWP.zip)
**Type** : bug
**NLog version**: 4.3.0-alpha3
**Platform**: UWP
**Current NLog config** (xml)

``` xml
<?xml version="1.0" encoding="utf-8" ?> 
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" 
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <targets>
    <target name="file" xsi:type="File" fileName="C:/Users/Dev/Documents/Project/myTestFile.txt" archiveEvery="Day"/>
  </targets>
  <rules>
    <logger name="*" minLevel="Trace" writeTo="file"/>
  </rules>
</nlog>
```

In case of a BUG:
- What is the current result?
  -> Currently no logs are written by either main logger or internalLogger
- What is the expected result?
  -> Log file is written to file myTestFile.txt
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
  ->Yes, although its not turned on in submitted code it was previously and that wasn't being written to either.
- Please post full exception details (message, stacktrace, inner exceptions)
  -> No exceptions are thrown.
- Are there any work arrounds? yes/no
  -> None found.
- Is there a version in which it did worked? 
  -> None found.
- Can you help us by writing an unit test?
  -> Probably not, depends on issue.

**Comment by 304NotModified** [comment]
Thanks! I will have a look to it

**Comment by 304NotModified** [comment]
FYI:

there are some small mistakes in the config, but even after fixing that it won't work. I will check more in-depth why not.

**Comment by 304NotModified** [comment]
Got this exception :(

```
System.InvalidOperationException: Synchronous operations should not be performed on the UI thread.  Consider wrapping this method in Task.Run.
   at System.IO.WinRTFileSystem.EnsureBackgroundThread()
   at System.IO.WinRTFileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
   at System.IO.MultiplexingWin32WinRTFileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean useAsync)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at NLog.Internal.FileAppenders.BaseFileAppender.TryCreateFileStream(Boolean allowFileSharedWriting)
   at NLog.Internal.FileAppenders.BaseFileAppender.CreateFileStream(Boolean allowFileSharedWriting)
   at NLog.Internal.FileAppenders.RetryingMultiProcessFileAppender.Write(Byte[] bytes)
   at NLog.Targets.FileTarget.WriteToFile(String fileName, LogEventInfo logEvent, Byte[] bytes, Boolean justData)
   at NLog.Targets.FileTarget.Write(LogEventInfo logEvent)
   at NLog.Targets.Target.Write(AsyncLogEventInfo logEvent)
```

**Comment by 304NotModified** [comment]
I looks like we have to do a lot before file writing will work in UWP. Sorry about that. Will try to fix it in 4.4

**Comment by karolys** [comment]
Thank you so much for taking a look. That's definitely strange but I know things are still pretty messy with UWP sdk so hopefully things get easier for you guys as MSFT irons it out. 

Looking forward to your fixes!

**Comment by snakefoot** [comment]
Have been able to use a NetStandard1_3 build of NLog 5.0 on a UWP-app both in debug and release-mode. See  #2055

**Comment by 304NotModified** [comment]
:+1:

**Comment by snakefoot** [comment]
@304NotModified Think this can be resolved with the release of NLog 5.0 BETA7

**Comment by 304NotModified** [comment]
yes indeed, thanks!
